KubeFed walkthrough ci check is flakey

<!-- Please only use this template for submitting reports about failing tests in KubeFed CI jobs -->

**Which test(s) are failing**:

KubeFed walkthrough check

**Since when has it been failing**:

Since merge of the check

**Reason for failure**:

```
echo 'Validating KubeFed walkthrough'
Validating KubeFed walkthrough
+./scripts/deploy-federated-nginx.sh
Validating KubeFed walkthrough.
Creating a test namespace test-namespace.
namespace/test-namespace created
Creating a ngnix application resources in namespace test-namespace.
configmap/web-file created
deployment.apps/nginx created
service/nginx created
Federating resources in namespace test-namespace to member clusters.
I0530 09:13:15.636692   16020 federate.go:451] Resource to federate is a namespace. Given namespace will itself be the container for the federated namespace
I0530 09:13:15.643778   16020 federate.go:480] Successfully created FederatedNamespace "test-namespace/test-namespace" from Namespace
I0530 09:13:15.655168   16020 federate.go:480] Successfully created FederatedConfigMap "test-namespace/web-file" from ConfigMap
I0530 09:13:15.660114   16020 federate.go:480] Successfully created FederatedDeployment "test-namespace/nginx" from Deployment
I0530 09:13:15.666141   16020 federate.go:480] Successfully created FederatedService "test-namespace/nginx" from Service
Checking status of federated resources.
Waiting for federated deployment status updated in cluster1. done
Waiting for federated deployment status updated in cluster2........................................................................................................................
[ERROR] Timeout waiting for federated deployment status updated in cluster2
```

**Anything else we need to know**:

The check is flakey - the failure is not consistent.  I've seen it 3 times this week, but we also haven't been running very many jobs.

<!-- DO NOT EDIT BELOW THIS LINE -->
/kind failing-test

--- a/scripts/deploy-federated-nginx.sh
+++ b/scripts/deploy-federated-nginx.sh
@@ -52,9 +52,9 @@ kubefedctl federate ns ${TEST_NS} --contents --skip-api-resources 'pods,secrets,
 echo
 echo "Checking status of federated resources."
 util::wait-for-condition 'federated deployment status updated in cluster1' \
-    "(kubectl get federateddeployment nginx -n ${TEST_NS} -o jsonpath='{.status.clusters[*].name}' | grep '^cluster1') &> /dev/null" 120
+    "(kubectl get federateddeployment nginx -n ${TEST_NS} -o jsonpath='{.status.clusters[*].name}' | grep '\<cluster1\>') &> /dev/null" 120
 util::wait-for-condition 'federated deployment status updated in cluster2' \
-    "(kubectl get federateddeployment nginx -n ${TEST_NS} -o jsonpath='{.status.clusters[*].name}' | grep '^cluster2') &> /dev/null" 120
+    "(kubectl get federateddeployment nginx -n ${TEST_NS} -o jsonpath='{.status.clusters[*].name}' | grep '\<cluster2\>') &> /dev/null" 120
 
 echo
 echo "Querying web server of deployment nginx application from member clusters."

